### PR TITLE
Add table azure_storage_sync. Closes #307

### LINKED ISSUE #307: Add table azure_storage_sync
**References**
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v55.4.0+incompatible/services/storagesync/mgmt/2020-03-01/storagesync

Please add private Endpoint Connections details.
Note: This detail is required for NIST SP 800 53 Rev 5 controls.

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro azure-test % ./tint.js azure_storage_sync
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_storage_sync []

PRETEST: tests/azure_storage_sync

TEST: tests/azure_storage_sync
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_resource_group.named_test_resource will be created
  + resource "azurerm_resource_group" "named_test_resource" {
      + id       = (known after apply)
      + location = "westeurope"
      + name     = "turbottest76647"
    }

  # azurerm_storage_sync.named_test_resource will be created
  + resource "azurerm_storage_sync" "named_test_resource" {
      + id                      = (known after apply)
      + incoming_traffic_policy = "AllowAllTraffic"
      + location                = "westeurope"
      + name                    = "turbottest76647"
      + resource_group_name     = "turbottest76647"
      + tags                    = {
          + "name" = "turbottest76647"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + region             = "westeurope"
  + resource_aka       = (known after apply)
  + resource_aka_lower = (known after apply)
  + resource_id        = (known after apply)
  + resource_name      = "turbottest76647"
  + subscription_id    = "d46d7416-f95f-4771-bbb5-529d4c76659c"
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 2s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest76647]
azurerm_storage_sync.named_test_resource: Creating...
azurerm_storage_sync.named_test_resource: Still creating... [10s elapsed]
azurerm_storage_sync.named_test_resource: Still creating... [20s elapsed]
azurerm_storage_sync.named_test_resource: Still creating... [30s elapsed]
azurerm_storage_sync.named_test_resource: Still creating... [40s elapsed]
azurerm_storage_sync.named_test_resource: Still creating... [50s elapsed]
azurerm_storage_sync.named_test_resource: Creation complete after 53s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest76647/providers/microsoft.storagesync/storageSyncServices/turbottest76647]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

region = "westeurope"
resource_aka = "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest76647/providers/microsoft.storagesync/storageSyncServices/turbottest76647"
resource_aka_lower = "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest76647/providers/microsoft.storagesync/storagesyncservices/turbottest76647"
resource_id = "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest76647/providers/microsoft.storagesync/storageSyncServices/turbottest76647"
resource_name = "turbottest76647"
subscription_id = "d46d7416-f95f-4771-bbb5-529d4c76659c"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest76647/providers/microsoft.storagesync/storageSyncServices/turbottest76647",
    "name": "turbottest76647",
    "region": "westeurope",
    "resource_group": "turbottest76647",
    "subscription_id": "d46d7416-f95f-4771-bbb5-529d4c76659c",
    "tags": {
      "name": "turbottest76647"
    },
    "type": "microsoft.storagesync/storageSyncServices"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest76647/providers/microsoft.storagesync/storageSyncServices/turbottest76647",
    "name": "turbottest76647",
    "region": "westeurope",
    "resource_group": "turbottest76647",
    "subscription_id": "d46d7416-f95f-4771-bbb5-529d4c76659c",
    "tags": {
      "name": "turbottest76647"
    },
    "type": "microsoft.storagesync/storageSyncServices"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest76647/providers/microsoft.storagesync/storageSyncServices/turbottest76647",
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest76647/providers/microsoft.storagesync/storagesyncservices/turbottest76647"
    ],
    "name": "turbottest76647",
    "tags": {
      "name": "turbottest76647"
    },
    "title": "turbottest76647"
  }
]
✔ PASSED

POSTTEST: tests/azure_storage_sync

TEARDOWN: tests/azure_storage_sync

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  id,
  type,
  provisioning_state
from
  azure_storage_sync;
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
| name              | id                                                                                                                                                                 | type             
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
| test-storage-sync | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/microsoft.storagesync/storageSyncServices/test-storage-sync | microsoft.storage
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
> select
  name,
  id,
  type,
  provisioning_state,
  incoming_traffic_policy
from
  azure_storage_sync
where
  incoming_traffic_policy = 'AllowVirtualNetworksOnly';
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
| name              | id                                                                                                                                                                 | type             
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
| test-storage-sync | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/microsoft.storagesync/storageSyncServices/test-storage-sync | microsoft.storage
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
> select
  name,
  id,
  connections ->> 'ID' as connection_id,
  connections ->> 'Name' as connection_name,
  connections ->> 'PrivateEndpointPropertyID' as property_private_endpoint_id,
  jsonb_pretty(connections -> 'PrivateLinkServiceConnectionState') as property_private_link_service_connection_state,
  connections ->> 'Type' as connection_type
from
  azure_storage_sync,
  jsonb_array_elements(private_endpoint_connections) as connections;
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
| name              | id                                                                                                                                                                 | connection_id    
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
| test-storage-sync | /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/arnab-test-resource-group/providers/microsoft.storagesync/storageSyncServices/test-storage-sync | /subscriptions/d4
|                   |                                                                                                                                                                    |                  
|                   |                                                                                                                                                                    |                  
|                   |                                                                                                                                                                    |                  
|                   |                                                                                                                                                                    |                  
+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
> 
```
</details>

### CODE REVIEW

File: azure/table_azure_storage_sync.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+// If we return the API response directly, the output will not provide all the properties of PrivateEndpointConnections
+func extractStorageSyncPrivateEndpointConnections(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	service := d.HydrateItem.(storagesync.Service)
+	info := []StoragePrivateEndpointConnections{}
+
+	if service.ServiceProperties.PrivateEndpointConnections != nil {
+		for _, connection := range *service.ServiceProperties.PrivateEndpointConnections {
+			properties := StoragePrivateEndpointConnections{}
+			properties.ID = connection.ID
```
Comment (ParthaI) [inline azure/table_azure_storage_sync.go:211]: Can we add the checks for top level property also?

File: docs/tables/azure_storage_sync.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+select
+  name,
+  id,
+  type,
+  provisioning_state
+from
+  azure_storage_sync;
+```
+
+### List storage sync which allows traffic only from private endpoints
```
Comment (ParthaI) [inline docs/tables/azure_storage_sync.md:19]:
```suggestion
### List storage sync which allows traffic through private endpoints only
```

File: azure/table_azure_storage_sync.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+				Name:        "subscription_id",
+				Description: ColumnDescriptionSubscription,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ID").Transform(idToSubscriptionID),
+			},
+		},
+	}
+}
+
+type StoragePrivateEndpointConnections struct {
```
Comment (ParthaI) [inline azure/table_azure_storage_sync.go:129]:
```suggestion
type StorageSyncPrivateEndpointConnections struct {
```

**Review by ParthaI — COMMENTED** [review]
@c0d3r-arnab please see the comments

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**